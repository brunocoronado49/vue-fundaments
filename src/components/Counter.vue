<template>
    <h2>{{ customTitle }}</h2>
    <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
    <div>
        <button @click="decrease">- 1</button>
        <button @click="increase">+ 1</button>
    </div>
</template>

<script>
export default {
    props: {
        msg: String,
        start: {
            // required: true,
            type: Number,
            default: 5,
            validator(value) {
                return value > 0
            }
        }
    },
    data() {
        return {
            counter: this.start
        }
    },
    methods: {
        getSquare() {
            return this.counter * this.counter
        },
        increase() {
            return this.counter++
        },
        decrease() {
            return this.counter--
        }
    },
    computed: {
        squareCounter() {
            return this.counter * this.counter
        }, 
        customTitle() {
            return this.msg || 'Counter'
        }
    }
}
</script>

<style>

button {
    margin: 0 5px;
    background-color: #64bb87;
    border: none;
    border-radius: 5px;
    color: white;
    font-weight: bold;
    cursor: pointer;
    padding: 5px 15px;
    transition: 0.3s ease-in-out;
}

button:hover {
    background-color: #5aa67b;
    transition: 0.3s ease-in-out;
}

</style>
