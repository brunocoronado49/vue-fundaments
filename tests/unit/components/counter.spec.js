import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter';

describe('Counter Component', () => {
    test('Must match with the snapshot', () => {
        const wrapper = shallowMount(Counter);
        
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('H2 must have default value "Counter"', () => {
        const wrapper = shallowMount(Counter);
        expect(wrapper.find('h2').exists()).toBeTruthy()
        const h2 = wrapper.find('h2').text();
        expect(h2).toBe('Counter');
    })
})
