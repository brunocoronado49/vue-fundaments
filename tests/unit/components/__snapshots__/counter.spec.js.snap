// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Must match with the snapshot 1`] = `
<h2>Counter</h2>
<p>5 <sup>2</sup> = 25</p>
<div class="buttons-container"><button>- 1</button><button>+ 1</button></div>
`;
